tree/ntupleutil/v7/src/RNTupleInspector.cxx

```diff
@@ -516,3 +516,34 @@ ROOT::Experimental::RNTupleInspector::GetFieldsByName(const std::regex &fieldNam
 
    return fieldIds;
 }
+
+void ROOT::Experimental::RNTupleInspector::PrintFieldsTreeAsDot(const ROOT::RFieldDescriptor &fieldDescriptor,
+                                                                std::ostream &output) const
+{
+   const auto &tupleDescriptor = GetDescriptor();
+   const bool isZeroField = fieldDescriptor.GetParentId() == ROOT::kInvalidDescriptorId;
+   if (isZeroField)
+      output << "digraph D {\n";
+   const std::string &nodeId = (isZeroField) ? "0" : std::to_string(fieldDescriptor.GetId() + 1);
+   const std::string &fieldName = (isZeroField) ? "RFieldZero" : fieldDescriptor.GetFieldName();
+   const std::string &description = fieldDescriptor.GetFieldDescription();
+
+   output << nodeId << "[label=<";
+   if (!isZeroField) {
+      output << "<b>Name: </b>" << fieldName << "<br></br>";
+      output << "<b>Type: </b>" << fieldDescriptor.GetTypeName() << "<br></br>";
+      output << "<b>ID: </b>" << std::to_string(fieldDescriptor.GetId()) << "<br></br>";
+      if (description != "")
+         output << "<b>Description: </b>" << fieldDescriptor.GetFieldDescription() << "<br></br>";
+      output << "<b>Version: </b>" << fieldDescriptor.GetFieldVersion() << "<br></br>";
+   } else
+      output << "<b>" << fieldName << "</b>";
+   output << ">, shape=box]\n";
```

Comment:
I'm pretty sure this `shape=box` can be made common rather than inline by adding one `node [shape=box]` at the beginning of the graph